spec/models/pacbio/well_spec.rb

```diff
@@ -84,4 +88,37 @@
     end
   end
 
+  context 'libraries' do
+    it 'can have one or more' do
+      well = create(:pacbio_well)
+      well.libraries << create_list(:pacbio_library, 5)
+      expect(well.libraries.count).to eq(5)
+    end
+  end
+
+  context 'request libraries' do
+
+    let(:well)                { create(:pacbio_well) }
+    let(:request_libraries)   { create_list(:pacbio_request_library, 2) }
+
+    before(:each) do
+      well.libraries << request_libraries.collect(&:library) 
+    end
+
+    it 'can have one or more' do
+      expect(well.request_libraries.length).to eq(2)
+    end
+
+    it 'can return a list of sample names' do
+      sample_names = well.sample_names.split(',')
+      expect(sample_names.length).to eq(2)
+      expect(sample_names.first).to eq(request_libraries.first.request.sample_name)
+    end
+
+    it 'can return a list of tags' do
+      expect(well.tags).to eq(request_libraries.collect(&:tag_id))
+    end
+
+  end
+
```

Comment:
tests for `sample_names` and `tags` methods

Reply:
The tests are there under the request_libraries context. Is that what you are asking?

Reply:
Ah yep sorry! Missed those